content/en/blog/2023/otel-end-user-discussions-february-2023.md

```diff
@@ -0,0 +1,187 @@
+
+**Q:** Are there any suggestions for monitoring OTel Collector's health or
+patterns for collecting agent telemetry?
+
+**A:** Collectors can be used to collect telemetry data from other collectors,
+which doesn't really need to be a disparate telemetry system. Users should also
+think about collecting multiple signals so that even if one signal fails, they
+get alerted by another. Here's an
+[article](https://ref.otel.help/otel-collector-ops/) discussing this.
+
+#### 2 - Timeline for opamp extension
+
+**Q:** Is there any timeline for implementing the opamp spec for agent
+management?
+
+**A:** It's not a top priority as of now. It would be good to have a maintainer
+from the community for opamp. You can track the issue
+[here](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16462).
+
+#### 3 - Buffer capabilities of OTel collector
+
+**Q:** What are some backup/retry buffer capabilities of OTel collectors when
+endpoints are unavailable?
+
+**A:** There is an experimental
+[storage extension](https://github.com/open-telemetry/opentelemetry-collector/tree/main/extension/experimental/storage)
+that is currently under development to support buffering and data persistence.
+
+### OpenTelemetry Language API & SDKs
+
+#### 1 - Timeline for Go SDK
+
+**Q:** What is the timeline for full specification compliance for OTel Go SDK?
+
+**A:** In the Go Otel SDK, the current progress is mostly around metrics. The
+logging development is frozen. Major work is being done in metrics sdk. Go
+metrics progress can be tracked
+[here](https://github.com/open-telemetry/opentelemetry-go/projects?query=is%3Aopen).
+Once metrics are done, logs will be taken care of.
+
+### OpenTelemetry Traces
+
+#### 1 - Sampling for traces
+
+**Q:** Is there a way to sample traces based on the span counts? Example:
+Drop/truncate traces which have more than 1000 spans in a single trace.
+
+**More context:** Sometimes, due to issues in the application itself, some
+traces generate a lot of spans that are not needed. Is there a way in
+OpenTelemetry to control this? More specifically, is there a way in which we can
+set up a condition where if a certain trace has more than ‘n’ number of spans,
+we can drop or truncate the number of spans.
+
+**A:** Tail-based sampling processor provides users with a bunch of sampling
+policies. Span count is one such policy. You can also combine multiple policies.
+Here's the link to tail
+[sampling processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/tailsamplingprocessor/README.md).
+The span count policy is based on min span count. Some users might look for some
+kind of exclusion policy.
+
+#### 2 - Use cases of span links
+
+**Q:** What are the use cases of span links?
+
+**A:** Span links are used to have an overview when there are thousands of
+spans. It helps in navigation, and answers questions like, "Where does a trace
+start and end?"
```

Comment:
That's not what span links are for, there are some use cases listed in the spec:

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md#links-between-spans

Reply:
And here: https://opentelemetry.io/docs/concepts/signals/traces/#span-links

I'm fine with having this in here for posterity, but we should probably note that this is not a correct answer before merging into a blog post.

Reply:
Hi @svrnm and @cartermp,

Thanks for the review. I have updated the description. Please review and let me know if it can be improved further.

Reply:
works for  me.